java/idea-ui/src/com/intellij/ide/projectWizard/generators/JdkDownloadService.kt

```diff
@@ -66,6 +66,7 @@ class JdkDownloadService(private val project: Project, private val coroutineScop
     val model = ProjectStructureConfigurable.getInstance(project).projectJdksModel
     coroutineScope.launch(Dispatchers.EDT) {
       writeAction {
+        model.syncSdks()
```

Comment:
I'll definitely add it as separate commit but this addresses the `SymbolicIdAlreadyExistsException` when basically it creates an entry for the downloaded JDK that already exists in the `jdk.table.xml`. The main reason is because the model isn't synchronized with the table keeping a reference to an old table state meaning that when tries to generate a new unique name fails.
```
com.intellij.platform.workspace.storage.impl.exceptions.SymbolicIdAlreadyExistsException: Entity with symbolicId: SdkId(name=openjdk-23 (4), type=JavaSDK) already exist
        at com.intellij.platform.workspace.storage.impl.MutableEntityStorageImpl.assertUniqueSymbolicId(ImmutableEntityStorageImpl.kt:282)
        at com.intellij.platform.workspace.storage.impl.MutableEntityStorageImpl.putEntity(ImmutableEntityStorageImpl.kt:249)
        at com.intellij.platform.workspace.storage.impl.ModifiableWorkspaceEntityBase.addToBuilder(Entities.kt:543)
        at com.intellij.platform.workspace.jps.entities.impl.SdkEntityImpl$Builder.applyToBuilder(SdkEntityImpl.kt:105)
        at com.intellij.platform.workspace.storage.impl.MutableEntityStorageImpl.addEntity(ImmutableEntityStorageImpl.kt:229)
        at com.intellij.workspaceModel.ide.impl.legacyBridge.sdk.SdkTableBridgeImpl.addNewSdk$lambda$5(SdkTableBridgeImpl.kt:79)
        at com.intellij.workspaceModel.ide.impl.GlobalWorkspaceModel.updateModel(GlobalWorkspaceModel.kt:114)
        at com.intellij.workspaceModel.ide.impl.legacyBridge.sdk.SdkTableBridgeImpl.addNewSdk(SdkTableBridgeImpl.kt:78)
        at com.intellij.openapi.projectRoots.impl.ProjectJdkTableImpl.addJdk(ProjectJdkTableImpl.kt:80)
        at com.intellij.ide.projectWizard.generators.JdkDownloadService$scheduleSetupInstallableSdk$1.invokeSuspend$lambda$2$lambda$0(JdkDownloadService.kt:70)
        at com.intellij.ide.projectWizard.generators.JdkDownloadService$scheduleSetupInstallableSdk$1.invokeSuspend$lambda$2$lambda$1(JdkDownloadService.kt:69)
        at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.lambda$createIncompleteSdk$9(ProjectSdksModel.java:502)
        at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.doAdd(ProjectSdksModel.java:549)
        at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.createIncompleteSdk(ProjectSdksModel.java:497)
        at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.setupInstallableSdk(ProjectSdksModel.java:477)
        at com.intellij.ide.projectWizard.generators.JdkDownloadService$scheduleSetupInstallableSdk$1.invokeSuspend$lambda$2(JdkDownloadService.kt:69)
```